Mm duplicate & quantity

Kiktamo opened the pull request:
Make changes to adding items to the cart to account for duplicates and have the cart items have a quantity property. Various changes to include the quantity in other calculations such as the cart total, subtotal, and cart price display. 

Can increase or decrease the quantity of items on the cart screen with them being removed when reaching zero.

Kiktamo committed:
Item quantities in the cart and assorted changes

Kiktamo committed:
format

Kiktamo committed:
Revert remove button, add plus and minus quantity

Kiktamo committed:
Format

joelcannon reviewed (changes requested):
function modifyQuantityInCart(productId, action) {
  const cartItems = getLocalStorage('so-cart');
  const existingItem = cartItems.find((item) => item.Id === productId);

  if (action === 'increase') {
    existingItem.quantity++;
  } else if (action === 'decrease') {
    existingItem.quantity--;
    if (existingItem.quantity <= 0) {
      removeProductFromCart(productId);
    }
  }

  setLocalStorage('so-cart', cartItems);
  updateCartCount();
}

// Usage:
// To increase quantity:
modifyQuantityInCart(productId, 'increase');

// To decrease quantity:
modifyQuantityInCart(productId, 'decrease');


function setupQuantityButtonListeners(buttons, increase = true) {
  buttons.forEach((button) => {
    button.addEventListener('click', (event) => {
      const productId = event.currentTarget.getAttribute('data-id');
      const action = increase ? 'increase' : 'decrease';
      modifyQuantityInCart(productId, action);
    });
  });
}

const increaseButtons = document.querySelectorAll('.increase-quantity');
const decreaseButtons = document.querySelectorAll('.decrease-quantity');

setupQuantityButtonListeners(increaseButtons, true); // For increase buttons
setupQuantityButtonListeners(decreaseButtons, false); // For decrease buttons

Kiktamo committed:
Make changes, implement quantity on product page

Kiktamo commented:
Alright I implemented the changes in the cart, and added a quantity in the product details page. It doesn't really use the same functions since what it does is different. The new quantity on the product details page changes how the add to cart button adds things. 

I did try to keep the actual code fairly minimal though. The product details also has a modify quantity function but I kept it to the single function design with increase or decrease being passed, I didn't use a setup function since unlike the cart there should only be one increase or decrease button per page anyway. Otherwise I believe it should function like you'd expect.

joelcannon reviewed (approved):
looks and works great.
